src/server.c

```diff
@@ -574,6 +574,54 @@ null_log_writer (GLogLevelFlags   log_level,
   return G_LOG_WRITER_HANDLED;
 }
 
+/* Bind server to available IPv4 and IPv6 local addresses
+ *
+ * The server must be NOT listening on any interface.
+ *
+ * If port is 0, the server will find an unsued port to listen on. The
+ * port will be the same for both addresses.
+ *
+ * Returns TRUE if the server was binded to at least one address. FALSE
+ * otherwise.
+ */
+static gboolean
+rstrnt_listen_any_local (SoupServer *server, guint port)
+{
+    GError   *error;
+    GSList   *uris;
+    gboolean  is_listening;
+
+    /* Ensure that server is not listening on any interface */
+    uris = soup_server_get_uris (server);
+    is_listening = g_slist_length (uris) > 0;
+    g_slist_free (uris);
+    g_return_val_if_fail (!is_listening, FALSE);
+
+    is_listening = soup_server_listen_local (server, port, SOUP_SERVER_LISTEN_IPV4_ONLY, &error);
+
+    if (!is_listening) {
+        g_warning ("Unable to listen on local IPv4 address: %s\n", error->message);
```

Comment:
Should test if error it not NULL before using.

Reply:
soup_server_listen_local sets error when FALSE is returned, but I agree is better to check the error instead.